service/worker/failoverManager/workflow.go

```diff
@@ -0,0 +1,376 @@
+
+	"github.com/uber/cadence/.gen/go/shared"
+	"github.com/uber/cadence/client/frontend"
+	"github.com/uber/cadence/common"
+)
+
+const (
+	failoverManagerContextKey = "failoverManagerContext"
+	// TaskListName tasklist
+	TaskListName = "cadence-sys-failoverManager-tasklist"
+	// WorkflowTypeName workflow type name
+	WorkflowTypeName = "cadence-sys-failoverManager-workflow"
+	// WorkflowID will be reused to ensure only one workflow running
+	WorkflowID             = "cadence-failover-manager"
+	failoverActivityName   = "cadence-sys-failover-activity"
+	getDomainsActivityName = "cadence-sys-getDomains-activity"
+
+	defaultBatchFailoverSize              = 10
+	defaultBatchFailoverWaitTimeInSeconds = 10
+
+	errMsgParamsIsNil                 = "params is nil"
+	errMsgTargetClusterIsEmpty        = "targetCluster is empty"
+	errMsgSourceClusterIsEmpty        = "sourceCluster is empty"
+	errMsgTargetClusterIsSameAsSource = "targetCluster is same as sourceCluster"
+
+	// QueryType for failover workflow
+	QueryType = "state"
+	// PauseSignal signal name for pause
+	PauseSignal = "pause"
+	// ResumeSignal signal name for resume
+	ResumeSignal = "resume"
+
+	// workflow states for query
+	wfRunning   = "running"
+	wfPaused    = "paused"
+	wfCompleted = "complete"
+)
+
+type (
+	// FailoverParams is the arg for failoverWorkflow
+	FailoverParams struct {
+		// TargetCluster is the destination of failover
+		TargetCluster string
+		// SourceCluster is from which cluster the domains are active before failover
+		SourceCluster string
+		// BatchFailoverSize is number of domains to failover in one batch
+		BatchFailoverSize int
+		// BatchFailoverWaitTimeInSeconds is the waiting time between batch failover
+		BatchFailoverWaitTimeInSeconds int
+		// Domains candidates to be failover
+		Domains []string
+	}
+
+	// FailoverResult is workflow result
+	FailoverResult struct {
+		SuccessDomains []string
+		FailedDomains  []string
+	}
+
+	// GetDomainsActivityParams params for activity
+	GetDomainsActivityParams struct {
+		TargetCluster string
+		SourceCluster string
+		Domains       []string
+	}
+
+	// FailoverActivityParams params for activity
+	FailoverActivityParams struct {
+		Domains       []string
+		TargetCluster string
+	}
+
+	// FailoverActivityResult result for failover activity
+	FailoverActivityResult struct {
+		SuccessDomains []string
+		FailedDomains  []string
+	}
+
+	// QueryResult ..
+	QueryResult struct {
+		TotalDomains   int
+		Success        int
+		Failed         int
+		State          string
+		TargetCluster  string
+		SourceCluster  string
+		SuccessDomains []string
+		FailedDomains  []string
+	}
+)
+
+func init() {
+	workflow.RegisterWithOptions(FailoverWorkflow, workflow.RegisterOptions{Name: WorkflowTypeName})
+	activity.RegisterWithOptions(FailoverActivity, activity.RegisterOptions{Name: failoverActivityName})
+	activity.RegisterWithOptions(GetDomainsActivity, activity.RegisterOptions{Name: getDomainsActivityName})
+}
```

Comment:
Those are deprecated? Should we use the new register?

Reply:
we should, I have plan to update all worker to use the new one later.